src/main/scala/io/qbeast/spark/utils/QbeastUtils.scala

```diff
@@ -28,54 +30,104 @@ import org.apache.spark.sql.DataFrame
 object QbeastUtils extends Logging {
 
   /**
-   * Compute the histogram for a given column
-   *
-   * Since computing the histogram can be expensive, this method is used outside the indexing
-   * process.
-   *
-   * It outputs the histogram of the column as format [bin1, bin2, bin3, ...] Number of bins by
-   * default is 50
-   *
-   * For example:
-   *
-   * val qbeastTable = QbeastTable.forPath(spark, "path")
-   *
-   * val histogram =qbeastTable.computeHistogramForColumn(df, "column")
-   *
-   * df.write.format("qbeast").option("columnsToIndex",
-   * "column:histogram").option("columnStats",histogram).save()
+   * Compute the quantiles for a given column of type String
    *
    * @param df
+   *   the DataFrame
    * @param columnName
+   *   the name of the column
    */
-  def computeHistogramForColumn(df: DataFrame, columnName: String, numBins: Int = 50): String = {
+  private def computeQuantilesForStringColumn(
+      df: DataFrame,
+      columnName: String,
+      numberOfQuantiles: Int): Array[String] = {
 
     import df.sparkSession.implicits._
-    if (!df.columns.contains(columnName)) {
-      throw AnalysisExceptionFactory.create(s"Column $columnName does not exist in the dataframe")
-    }
-
-    log.info(s"Computing histogram for column $columnName with number of bins $numBins")
     val binStarts = "__bin_starts"
     val stringPartitionColumn =
-      MultiDimClusteringFunctions.range_partition_id(col(columnName), numBins)
+      MultiDimClusteringFunctions.range_partition_id(col(columnName), numberOfQuantiles)
 
-    val histogram = df
+    val quantiles = df
       .select(columnName)
-      .distinct()
-      .na
```

Comment:
Is it ok to drop "null"s?

Reply:
good catch, I deleted unnecessary code lines 🤡 

I think that to mimic the behaviour of `approxQuantiles` for number, we should leave null and nans out of the calculation